### PR TITLE
Feature/aruha 130

### PR DESCRIPTION
**Pull request by jkliff**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/zalando/nakadi/pull/170" target="_blank">ReviewNinja

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 1]
```diff
@@ -255,14 +269,14 @@ public void whenCreateSuccessfullyThen201() throws Exception {
         Mockito
                 .doNothing()
                 .when(topicRepository)
-                .createTopic("event-name");
+                .createTopic("team.event-name");
```
Comment (CyberDem0n) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: .createTopic(EVENT_TYPE_NAME)

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 2]
```diff
@@ -255,14 +269,14 @@ public void whenCreateSuccessfullyThen201() throws Exception {
...
 
         postEventType(buildEventType())
                 .andExpect(status().isCreated())
                 .andExpect(content().string(""));
 
         verify(eventTypeRepository, times(1)).saveEventType(any(EventType.class));
-        verify(topicRepository, times(1)).createTopic("event-name");
+        verify(topicRepository, times(1)).createTopic("team.event-name");
```
Comment (CyberDem0n) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: .createTopic(EVENT_TYPE_NAME)

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 3]
```diff
@@ -275,23 +289,23 @@ public void whenTopicCreationFailsRemoveEventTypeFromRepositoryAnd500() throws E
...
-                .andExpect(content().contentType("application/problem+json"))
-                .andExpect(content().string(matchesProblem(expectedProblem)));
+            .andExpect(status().isServiceUnavailable())
+            .andExpect(content().contentType("application/problem+json"))
+            .andExpect(content().string(matchesProblem(expectedProblem)));
 
         verify(eventTypeRepository, times(1)).saveEventType(any(EventType.class));
-        verify(topicRepository, times(1)).createTopic("event-name");
-        verify(eventTypeRepository, times(1)).removeEventType("event-name");
+        verify(topicRepository, times(1)).createTopic("team.event-name");
```
Comment (CyberDem0n) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: .createTopic(EVENT_TYPE_NAME)

File: src/main/java/de/zalando/aruha/nakadi/domain/EventType.java [hunk 4]
```diff
@@ -16,7 +17,7 @@
     public static final List<String> EMPTY_STRING_LIST = new ArrayList<>(0);
 
     @NotNull
-    @Size(min = 1, message = "may not be empty")
+    @Pattern(regexp = "[-0-9a-zA-Z_]+(\\.[-0-9a-zA-Z_]+)*", message = "format not allowed" )
```
Comment (CyberDem0n) [inline src/main/java/de/zalando/aruha/nakadi/domain/EventType.java]: regexp doesn't match with pattern from swagger.yml

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 5]
```diff
@@ -167,34 +183,39 @@ public void whenPostAndTopicExistsReturn409() throws Exception {
 
     @Test
     public void whenDeleteEventTypeThenOk() throws Exception {
+
+final String eventTypeName = TestUtils.randomValidEventTypeName();
```
Comment (rcillo) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: This needs some indentation.

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 6]
```diff
@@ -167,34 +183,39 @@ public void whenPostAndTopicExistsReturn409() throws Exception {
...
-        verify(eventTypeRepository, times(1)).removeEventType(EVENT_TYPE_NAME);
-        verify(topicRepository, times(1)).deleteTopic(EVENT_TYPE_NAME);
+        verify(eventTypeRepository, times(1)).removeEventType(eventTypeName);
+        verify(topicRepository, times(1)).deleteTopic(eventTypeName);
     }
 
     @Test
     public void whenDeleteNoneExistingEventTypeThen404() throws Exception {
+
+final String eventTypeName = TestUtils.randomValidEventTypeName();
```
Comment (rcillo) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: As well.

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 7]
```diff
@@ -202,14 +223,16 @@ public void whenDeleteNoneExistingEventTypeThen404() throws Exception {
 
     @Test
     public void whenDeleteEventTypeAndTopicDeletionExceptionThen503() throws Exception {
+
+final String eventTypeName = TestUtils.randomValidEventTypeName();
```
Comment (rcillo) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: as well.

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 8]
```diff
@@ -217,14 +240,16 @@ public void whenDeleteEventTypeAndTopicDeletionExceptionThen503() throws Excepti
 
     @Test
     public void whenDeleteEventTypeAndNakadiExceptionThen500() throws Exception {
+
+final String eventTypeName = TestUtils.randomValidEventTypeName();
```
Comment (rcillo) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: here too.

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 9]
```diff
@@ -255,18 +282,21 @@ public void whenCreateSuccessfullyThen201() throws Exception {
...
 
         verify(eventTypeRepository, times(1)).saveEventType(any(EventType.class));
-        verify(topicRepository, times(1)).createTopic("event-name");
+        verify(topicRepository, times(1)).createTopic(et.getName());
     }
 
     @Test
     public void whenTopicCreationFailsRemoveEventTypeFromRepositoryAnd500() throws Exception {
+
+//final String eventTypeName = TestUtils.randomValidEventTypeName();
```
Comment (rcillo) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: I think you can remove this comment and indent the line bellow.

File: src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java [hunk 10]
```diff
@@ -400,13 +434,15 @@ public void canExposeASingleEventType() throws Exception {
 
     @Test
     public void askingForANonExistingEventTypeResultsIn404() throws Exception {
-        when(eventTypeRepository.findByName(anyString())).thenThrow(new NoSuchEventTypeException("EventType 'event-name' does not exist."));
+    	final String eventTypeName = TestUtils.randomValidEventTypeName();
```
Comment (rcillo) [inline src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java]: Indentation.

**Comment by rcillo** [comment]
Nice cleanup.

Just have minor formats regarding indentation.

Besides that, :shipit:

**Pull request merged**